pkg/controller/appdefinition/volume.go

```diff
@@ -206,6 +209,55 @@ func toPVCs(req router.Request, appInstance *v1.AppInstance) (result []kclient.O
 	return
 }
 
+func GetPVForVolumeBinding(req router.Request, appInstance *v1.AppInstance, binding v1.VolumeBinding) (*corev1.PersistentVolume, error) {
+	// binding.Volume can either be the actual name of the PersistentVolume, or its public name in Acorn.
+	// Check for the actual name first.
+	pv := new(corev1.PersistentVolume)
+	if err := req.Client.Get(req.Ctx, kclient.ObjectKey{Name: binding.Volume}, pv); err != nil && !apierrors.IsNotFound(err) {
+		return nil, err
+	} else if err == nil {
+		// make sure this PV is managed by acorn
+		if _, ok := pv.Labels[labels.AcornManaged]; !ok {
+			return nil, apierrors.NewNotFound(schema.GroupResource{Group: "", Resource: "persistentvolumes"}, binding.Volume)
```

Comment:
Returning not-found errors here could be confusing for a user.

Let's say someone is running Acorn on a cluster they have admin controls over. If they create a volume and then try to bind said volume, they would see a status indicating that the volume was not found. However, they would be able to see that the volume would exist, and, likely, create an issue with Acorn.

Could this error be changed to indicate that no "acorn managed" volume was found? Does the error returned here need to be a Kubernetes API error?

I have the same comments/questions about the other errors. Obviously, the error needn't be too descriptive, but something other than a plain "not found" would be better, in my opinion.

Reply:
Yeah that sounds like a good idea. My rationale for plain NotFound errors was that I did not want to indicate whether a PV existed with the provided name, as ordinary users would be able to abuse that to try to guess names of other volumes that exist in the cluster (and also the names of other apps). But returning a more Acorn-specific error sounds good as long as I am still properly obscuring that information.